web/yo/app/scripts/services/pubUtils.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+/**
+ * @ngdoc factory
+ * @name oncokbApp.pubUtils
+ * @author Jing Su
+ * @description
+ * This factory is a cache service used for storing validated PMIDs.
+ */
+angular.module('oncokbApp')
+    .factory('pubUtils', function() {
+        var validatedPubs = {};
```

Comment:
Do we need to cache the full content of the PMID?

Reply:
The full content of PMID are `type, id and valid`. If you think the data is too much, I can only store `valid`.

Reply:
Oh right, you also want to cache . the invalid as well. Cool, thanks for the info.